lib/StaticUniformGrid.js

```diff
@@ -0,0 +1,277 @@
+
+    /**
+     * Start index for each cell's data within the data array.
+     * @type {Number[]}
+     */
+    this.cellIndices = [];
+
+    /**
+     * Count for each cell's data within the data array.
+     * @type {Number[]}
+     */
+    this.cellCounts = [];
+
+    /**
+     * Array of objects
+     * @type {Array}
+     */
+    this.data = [];
+
+    ////// Populate //////
+    var min = this.AABB.minimum;
+    var max = this.AABB.maximum;
+    var center = this.AABB.center;
+    min.x = Number.POSITIVE_INFINITY;
+    min.y = Number.POSITIVE_INFINITY;
+    min.z = Number.POSITIVE_INFINITY;
+    max.x = Number.NEGATIVE_INFINITY;
+    max.y = Number.NEGATIVE_INFINITY;
+    max.z = Number.NEGATIVE_INFINITY;
+
+    var resolution = this.resolution;
+
+    // Find the min/max bounds and resolution of the uniform grid
+    var objectCount = data.length;
+    for (var i = 0; i < objectCount; i++) {
+        objectAABBFunction(data[i], min, max);
+    }
+
+    // Figure out what the grid's resolution should be. Pad min and max out to match
+    resolution.x = Math.floor((max.x - min.x) / cellWidth) + 1;
+    resolution.y = Math.floor((max.y - min.y) / cellWidth) + 1;
+    resolution.z = Math.floor((max.z - min.z) / cellWidth) + 1;
+
+    center = Cartesian3.add(min, max, center);
+    center = Cartesian3.divideByScalar(center, 2.0, center);
+    min.x = center.x - (resolution.x / 2.0) * cellWidth;
+    min.y = center.y - (resolution.y / 2.0) * cellWidth;
+    min.z = center.z - (resolution.z / 2.0) * cellWidth;
+    max.x = center.x + (resolution.x / 2.0) * cellWidth;
+    max.y = center.y + (resolution.y / 2.0) * cellWidth;
+    max.z = center.z + (resolution.z / 2.0) * cellWidth;
+
+    var cellCount = resolution.x * resolution.y * resolution.z;
+
+    // Bin the objects
+    var allCellData = [];
+    for (i = 0; i < cellCount; i++) {
+        allCellData.push({
+            itemCount : 0,
+            items : []
+        });
+    }
+
+    var parameters = {
+        allCellData : allCellData,
+        cellCheckFunction : cellCheckFunction,
+        grid : this,
+        item : undefined
+    };
+
+    //For each object:
+    for (i = 0; i < objectCount; i++) {
+        var item = data[i];
+
+        // Get the object's AABB
+        minScratch.x = Number.POSITIVE_INFINITY;
+        minScratch.y = Number.POSITIVE_INFINITY;
+        minScratch.z = Number.POSITIVE_INFINITY;
+        maxScratch.x = Number.NEGATIVE_INFINITY;
+        maxScratch.y = Number.NEGATIVE_INFINITY;
+        maxScratch.z = Number.NEGATIVE_INFINITY;
+        objectAABBFunction(item, minScratch, maxScratch);
+
+        // Step over the cells in the AABB, checking for each cell if the object intersects this cell
+        parameters.item = item;
+        StaticUniformGrid.boundingBoxMarch(this, minScratch, maxScratch, binObject, parameters);
+    }
+
+    // Store all the object copies in one contiguous array for better spatial locality
```

Comment:
I'm not sure that this is true; it depends on access patterns.  You would want a space fitting curve here.  Could be interesting future work.